source/framework/include/marlin/ProcessorApi.h

```diff
@@ -56,45 +51,203 @@ namespace marlin {
         | book::Flags::value(book::Flags::Book::Store)};
 
       /**
-       *  @brief Register new Histogram.
-       *  @param proc the processor instance
-       *  @param path to registered histogram
-       *  @param axes array of axis configuration for the Histogram 
-       *  @throw TODO: ErrorType if an Object at path already exist with other type
+       *  @brief  Book  a histogram 1D, float type
+       *  
+       *  @param  proc       the processor booking the histogram
+       *  @param  path       the histogram entry path
+       *  @param  name       the histogram name
+       *  @param  title      the histogram title
+       *  @param  axisconfig the histogram X axis config
+       *  @param  flags      the book flag policy
        */
-      template<typename HistT>
-      static  book::Handle<book::Entry<HistT>>
-      create(
-        const Processor * const proc,
-        const std::filesystem::path& pathName,
-        const std::string_view& title,
-        const std::array<
-          book::AxisConfig<typename HistT::Precision_t>,
-          HistT::Dimension>& axes,
-        const book::Flag_t& flags = DefaultConfiguration ) ;
+      [[nodiscard]] static H1FEntry bookHist1F (
+        const Processor *proc, 
+        const std::filesystem::path &path, 
+        const std::string_view &name,
+        const std::string_view &title,
+        const AxisConfigD &axisconfig,
+        const BookFlag &flags  = DefaultConfiguration) ; 
 
       /**
-       *  @brief Get Handle to existing histogram entry.
-       *  Histograms can registered in steering file or with registerHistogram() 
-       *  @param proc the processor instance
-       *  @param path to registered histogram
-       *  @throw TODO: ErrorType if path don't belong to an Object.
+       *  @brief  Book  a histogram 2D, float type
+       *  
+       *  @param  proc        the processor booking the histogram
+       *  @param  path        the histogram entry path
+       *  @param  name        the histogram name
+       *  @param  title       the histogram title
+       *  @param  axisconfigX the histogram X axis configuration
+       *  @param  axisconfigY the histogram Y axis configuration
+       *  @param  flags       the book flag policy
+       */
+      [[nodiscard]] static H2FEntry bookHist2F (
+        const Processor *proc, 
+        const std::filesystem::path &path, 
+        const std::string_view &name,
+        const std::string_view &title,
+        const AxisConfigD &axisconfigX,
+        const AxisConfigD &axisconfigY,
+        const BookFlag &flags  = DefaultConfiguration) ; 
+
```

Comment:
Is `bookHist3F()` missing on purpose ?

Reply:
no, sorry, just miss it, will add H1I, H2I and H3I also